plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/IncrementPropertyAction.java

```diff
@@ -0,0 +1,114 @@
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+
+public class IncrementPropertyAction implements ActionExecutor {
+    private static final Logger logger = LoggerFactory.getLogger(IncrementPropertyAction.class.getName());
+
+    @Override
+    public int execute(final Action action, final Event event) {
+        boolean storeInSession = Boolean.TRUE.equals(action.getParameterValues().get("storeInSession"));
+        if (storeInSession && event.getSession() == null) {
+            return EventService.NO_CHANGE;
+        }
+
+        String propertyName = (String) action.getParameterValues().get("propertyName");
+        Profile profile = event.getProfile();
+        Session session = event.getSession();
+
+        try {
+            Map<String, Object> properties = storeInSession ? session.getProperties() : profile.getProperties();
+            Object propertyValue = getPropertyValue(action, event, propertyName, properties);
+
+            if (PropertyHelper.setProperty(properties, propertyName, propertyValue, "alwaysSet")) {
+                return storeInSession ? EventService.SESSION_UPDATED : EventService.PROFILE_UPDATED;
+            }
+        } catch (IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+            logger.warn("Error resolving nested property of object. See debug log level for more information");
+            if (logger.isDebugEnabled()) {
+                logger.error("Error resolving nested property of item: {}", storeInSession ? session : profile, e);
+            }
+        }
+
+        return EventService.NO_CHANGE;
+    }
+
+    private Object getPropertyValue(Action action, Event event, String propertyName, Map<String, Object> properties)
+            throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        String propertyTarget = (String) action.getParameterValues().get("propertyTarget");
+        String rootPropertyName = propertyName.split("\\.")[0];
+        Object propertyValue = 1;
+
+        Object propertyTargetValue = null;
+
+        if (StringUtils.isNotEmpty(propertyTarget)) {
+            propertyTargetValue = PropertyUtils.getNestedProperty(((CustomItem) event.getTarget()).getProperties(), propertyTarget);
+        }
+
+        if (propertyTargetValue != null) {
+            if (propertyTargetValue instanceof Integer) {
+                if (properties.containsKey(rootPropertyName)) {
+                    propertyValue = (int) propertyTargetValue + (int) PropertyUtils.getNestedProperty(properties, propertyName);
+                } else {
+                    propertyValue = propertyTargetValue;
+                }
+            } else if (propertyTargetValue instanceof Map) {
+                if (properties.containsKey(rootPropertyName)) {
+                    Map<String, Object> nestedProperty = (Map<String, Object>) PropertyUtils.getNestedProperty(properties, propertyName);
```

Comment:
We should check that PropertyUtils.getNestedProperty(properties, propertyName); is actually not null and it's a map, if not log an error and continue.